### PR TITLE
[Mission] CoP5-3 Mechanics

### PR DESCRIPTION
**Pull request by The-Aerec**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

Adds mob mechanics, resistances, abilities, spells to BCNMs pertaining to CoP5-2: Three paths.

## Steps to test these changes

!togglegm
Head to any of the three BCNM fights and test the BCNMS for:

Head Wind
Flames for the Dead
A Century of Hardship

### CODE REVIEW

File: sql/mob_pool_mods.sql [hunk 1]
```diff
@@ -59,6 +59,9 @@ INSERT INTO `mob_pool_mods` VALUES (459,73,25,0); -- (Bloodlapper) Store TP
 INSERT INTO `mob_pool_mods` VALUES (459,430,20,0); -- (Bloodlapper) Quad Attack
 INSERT INTO `mob_pool_mods` VALUES (519,160,-50,0);
 INSERT INTO `mob_pool_mods` VALUES (532,160,-50,0);
+INSERT INTO `mob_pool_mods` VALUES (559,62,-50,0); -- Bugbby -ATTP-
+INSERT INTO `mob_pool_mods` VALUES (559,288,40,100); --DATT
```
Comment (claywar) [inline sql/mob_pool_mods.sql]: CI is failing here and the line below, space is required between `--` for comment and text afterwards

File: scripts/globals/items/chunk_of_shumeyo_salt.lua [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+require("scripts/globals/player")
+local ID = require("scripts/zones/Bearclaw_Pinnacle/IDs")
+-----------------------------------------
+local item_object = {}
+
+item_object.onItemCheck= function(target, player)
+    local result = 0
+	local id = target:getID()
+    local checkID = true
+    local snollID =
```
Comment (claywar) [inline scripts/globals/items/chunk_of_shumeyo_salt.lua]: These IDs should probably be an offset in IDs.lua for the zone

File: scripts/globals/mobskills/arctic_impact.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+require("scripts/globals/mobskills")
+---------------------------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+
```
Comment (claywar) [inline scripts/globals/mobskills/arctic_impact.lua]: Can remove the whitespace at the beginning and end of this function

File: scripts/globals/mobskills/arctic_impact.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---------------------------------------------------
+-- Arctic Impact
+-- Deals Ice damage to enemies within range.
+-- Area of Effect is centered around caster.
+---------------------------------------------------
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/arctic_impact.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/dancing_edge.lua [hunk 5]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Dancing Edge
+-- Delivers a fivefold attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/dancing_edge.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/evisceration.lua [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Evisceration
+-- Delivers a fivefold attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/evisceration.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/hiemal_storm.lua [hunk 7]
```diff
@@ -0,0 +1,30 @@
+---------------------------------------------
+--  Hiemal Storm
+--
+--  Description: Extreme directional AoE ice damage for 200-1400 points
+--  Type: Breath
+--  Utsusemi/Blink absorb: Ignores shadows
+--  Range: Unknown cone
+--  Notes: Used only by Snoll Tzar
+---------------------------------------------
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/hiemal_storm.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/impulse_drive.lua [hunk 8]
```diff
@@ -0,0 +1,39 @@
+---------------------------------------------
+-- Impulse Drive
+-- Delivers a two-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/impulse_drive.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/penta_thrust.lua [hunk 9]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Penta Thrust
+-- Delivers a five-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/penta_thrust.lua:8]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/shadowstitch.lua [hunk 10]
```diff
@@ -0,0 +1,39 @@
+---------------------------------------------
+-- Shadowstitch
+-- Binds target.
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/shadowstitch.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/skewer.lua [hunk 11]
```diff
@@ -0,0 +1,39 @@
+---------------------------------------------
+-- Skewer
+-- Delivers a three-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/skewer.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/spiral_hell.lua [hunk 12]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Spiral Hell
+-- Delivers a single-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/spiral_hell.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/vorpal_scythe.lua [hunk 13]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Vorpal Scythe
+-- Delivers a single-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/vorpal_scythe.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/globals/mobskills/wheeling_thrust.lua [hunk 14]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------
+-- Wheeling Thrust
+-- Delivers a two-hit attack
+-- Type: Physical
+-- Range: Melee
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/wheeling_thrust.lua]: It doesn't look like any settings are used in this file, so this require isn't needed.

File: scripts/zones/Bearclaw_Pinnacle/IDs.lua [hunk 15]
```diff
@@ -17,9 +17,13 @@ zones[xi.zone.BEARCLAW_PINNACLE] =
         CARRIED_OVER_POINTS       = 7001, -- You have carried over <number> login point[/s].
         LOGIN_CAMPAIGN_UNDERWAY   = 7002, -- The [/January/February/March/April/May/June/July/August/September/October/November/December] <number> Login Campaign is currently underway!<space>
         LOGIN_NUMBER              = 7003, -- In celebration of your most recent login (login no. <number>), we have provided you with <number> points! You currently have a total of <number> points.
-        PARTY_MEMBERS_HAVE_FALLEN = 7403, -- All party members have fallen in battle. Now leaving the battlefield.
+        PARTY_MEMBERS_HAVE_FALLEN = 7406, -- All party members have fallen in battle. Now leaving the battlefield.
```
Comment (claywar) [inline scripts/zones/Bearclaw_Pinnacle/IDs.lua]: Need to update these IDs for current client.  This is actually wrong previously, and as of May is 7407.  The other additions are also incorrect (Blown away is 7615, for example)

File: scripts/zones/Mine_Shaft_2716/mobs/Bugbby.lua [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+require("scripts/globals/status")
+mixins = {require("scripts/mixins/job_special")}
+local ID = require("scripts/zones/Mine_Shaft_2716/IDs")
+require("scripts/globals/status")
+-----------------------------------
+local entity = {}
+
+local aCenturyOfHardship =
+{
+    [1] =
```
Comment (claywar) [inline scripts/zones/Mine_Shaft_2716/mobs/Bugbby.lua]: Any raw ID should be referenced from the IDs.lua file (enum, and pull them into tables here if needed, that way if there are shifts, its contained to a subset of files)

File: scripts/zones/Bearclaw_Pinnacle/IDs.lua (hunk 15 above)
Comment (The-Aerec) [reply scripts/zones/Bearclaw_Pinnacle/IDs.lua]: I was on the most up to date client when I checked this o.o. the texts came out just fine.

Comment (The-Aerec) [reply scripts/zones/Bearclaw_Pinnacle/IDs.lua]: But i'll double check

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]
Styling issues:
- Logic in several lines could be separated in different lines.
```
if X then
    Y
end
```
Alignment in this lines, among others is wrong. 3 spaces instead of 4
- chunk_of_shumeyo_salt.lua: line 15 and 24
- dancing_edge.lua: line 26
- evisceration: line 26
- impulsive_drive.lua: line 34
- penta_thrust.lua: line 27
- shadowstich.lua: line 34
- skewer.lua: line 34
- spiral_hell.lua: line 26

and so on

**Comment by The-Aerec** [comment]
> Styling issues:
> 
>     * Logic in several lines could be separated in different lines.
> 
> 
> ```
> if X then
>     Y
> end
> ```
> 
> Alignment in this lines, among others is wrong. 3 spaces instead of 4
> 
>     * chunk_of_shumeyo_salt.lua: line 15 and 24
> 
>     * dancing_edge.lua: line 26
> 
>     * evisceration: line 26
> 
>     * impulsive_drive.lua: line 34
> 
>     * penta_thrust.lua: line 27
> 
>     * shadowstich.lua: line 34
> 
>     * skewer.lua: line 34
> 
>     * spiral_hell.lua: line 26
> 
> 
> and so on

Addressed

File: scripts/zones/Mine_Shaft_2716/mobs/Bugbby.lua [hunk 17]
```diff
@@ -0,0 +1,56 @@
+-----------------------------------
+-- Area: Mine Shaft 2716
+-- CoP Mission 5-3 (A Century of Hardship)
+-- NM: Bugbby
+-----------------------------------
+require("scripts/globals/status")
+mixins = {require("scripts/mixins/job_special")}
+local ID = require("scripts/zones/Mine_Shaft_2716/IDs")
+require("scripts/globals/status")
```
Comment (Xaver-DaRed) [inline scripts/zones/Mine_Shaft_2716/mobs/Bugbby.lua:7]: I don't think we ever internally got to a consesus about this, but i think we generaly put local ID first, then all requires toguether and then mixins

except in the interaction system, where we put local ID rquire last thing.

File: scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua [hunk 18]
```diff
@@ -2,9 +2,92 @@
...
+           mob:getBattleTime() - changeTime > 11
+    then
+        mob:setLocalVar("delayed", 0)
+        mob:setAnimationSub(7)
+        mob:setLocalVar("changeTime", mob:getBattleTime())
+        mob:setDamage(160)
+    -- self-destruct
+    elseif delay < os.time() and
+           mob:getAnimationSub() == 7 and
+           mob:getBattleTime() - changeTime > 12
```
Comment (Xaver-DaRed) [inline scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua]:
```
elseif
    delay <os.time() and
    The_Rest
then
```

File: scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua [hunk 19]
```diff
@@ -2,9 +2,92 @@
...
+           mob:getAnimationSub() == 5 and
+           mob:getBattleTime() - changeTime > 11
+    then
+        player:messageText(player, ID.text.LARGE_STEAM) -- approx. midway point - give warning
+        mob:setLocalVar("delayed", 0)
+        mob:setAnimationSub(6)
+        mob:setLocalVar("changeTime", mob:getBattleTime())
+        mob:setDamage(150)
+    -- biggest
+    elseif delay < os.time() and
```
Comment (Xaver-DaRed) [inline scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua]:
```
elseif
    delay <os.time() and
    The_Rest
then
```

File: scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua [hunk 20]
```diff
@@ -2,9 +2,92 @@
...
+    if delay < os.time() and
+       mob:getAnimationSub() == 4 and
+       mob:getBattleTime() - changeTime > 11
+    then
+        mob:setLocalVar("delayed", 0)
+        mob:setAnimationSub(5)
+        mob:setLocalVar("changeTime", mob:getBattleTime())
+        mob:setDamage(140)
+    -- bigger
+    elseif delay < os.time() and
```
Comment (Xaver-DaRed) [inline scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua]:
```
elseif
    delay <os.time() and
    The_Rest
then
```

File: scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua [hunk 21]
```diff
@@ -2,9 +2,92 @@
...
+    -- handle salt cooldown
+    if cd < os.time() and
+       salty == 1
+    then
+        player:messageText(player, ID.text.SHOOK_SALT)
+        mob:setLocalVar("salty", 0)
+    end
+
+    -- big
+    if delay < os.time() and
```
Comment (Xaver-DaRed) [inline scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua]:
```
if
    delay <os.time() and
    The_Rest
then
```

File: scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua [hunk 22]
```diff
@@ -2,9 +2,92 @@
...
+
+    mob:setDamage(130)
+
+    -- handle salt usage
+    if melting == 1 then
+        mob:setLocalVar("melt", 0)
+    end
+
+    -- handle salt cooldown
+    if cd < os.time() and
```
Comment (Xaver-DaRed) [inline scripts/zones/Bearclaw_Pinnacle/mobs/Snoll_Tzar.lua]:
```
elseif
    cd <os.time() and
    The_Rest
then
```

File: scripts/globals/mobskills/vorpal_scythe.lua [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local numhits = 1
+    local accmod = 1
+    local dmgmod = 1
+    local info = xi.mobskills.mobPhysicalMove(mob, target, skill, numhits, accmod, dmgmod, 3, 1, 1, 1)
+    local dmg = xi.mobskills.mobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.SLASHING, info.hitslanded)
+
+    target:takeDamage(dmg, mob, xi.attackType.PHYSICAL, xi.damageType.SLASHING)
+
+	if dmg > 0 and skill:getMsg() ~= 31 then target:tryInterruptSpell(mob, info.hitslanded) end
```
Comment (Xaver-DaRed) [inline scripts/globals/mobskills/vorpal_scythe.lua]: theres seems to be 5 spaces in this line
could also separate lines

**Review by Xaver-DaRed — DISMISSED** [review]

File: scripts/globals/mobskills/evisceration.lua [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+    local info = xi.mobskills.mobPhysicalMove(mob, target, skill, numhits, accmod, dmgmod, 3, 1, 1, 1)
+    local dmg = xi.mobskills.mobFinalAdjustments(info.dmg, mob, skill, target, xi.attackType.PHYSICAL, xi.damageType.PIERCING, info.hitslanded)
+
+    target:takeDamage(dmg, mob, xi.attackType.PHYSICAL, xi.damageType.PIERCING)
+
+    if
+        dmg > 0 and
+        skill:getMsg() ~= 31
+    then
+    target:tryInterruptSpell(mob, info.hitslanded)
```
Comment (claywar) [inline scripts/globals/mobskills/evisceration.lua]: indent needed here

**Review by claywar — APPROVED** [review]

**Review by TeoTwawki — APPROVED** [review]

**Pull request merged**